### PR TITLE
Adds support for Guzzle 6

### PR DESCRIPTION
**Pull request by vicdelfant**
This PR adds support for Guzzle 6 (see #17). All tests pass except for one which is kind of hard to test manually. The updated version of the library seems to work just fine in our production environment.

```
There was 1 failure:

1) ApiTest::testPhoneVerificationCheck
Failed asserting that false matches expected true.
```

Note that I didn't see a consistent coding style (single quote vs. double quote and `array` vs. `[]`) so I went with a single quote and left the array initialization as-is.

### CODE REVIEW

File: lib/Authy/AuthyApi.php [hunk 1]
```diff
@@ -41,12 +41,9 @@ class AuthyApi
     public function __construct($api_key, $api_url = "https://api.authy.com")
     {
         $this->rest = new \GuzzleHttp\Client(array(
-            'base_url' => "{$api_url}/protected/json/",
-            'defaults' => array(
-                'headers'       => array('User-Agent' => $this->__getUserAgent() ),
-                'query'         => array('api_key' => $api_key),
```
Comment (dcu) [inline lib/Authy/AuthyApi.php]: why did you remove this default option?

Comment (vicdelfant) [inline lib/Authy/AuthyApi.php]: Guzzle 6 no longer uses the `defaults` subarray, to set defaults you simply pass the headers as a root element: http://guzzle.readthedocs.org/en/latest/request-options.html#headers.

Comment (dcu) [inline lib/Authy/AuthyApi.php]: According to this it's supported, did you test it?
https://github.com/guzzle/guzzle/blob/master/src/Client.php#L56

Comment (vicdelfant) [inline lib/Authy/AuthyApi.php]: Yes, it was tested. I don't see anything in Client.php in the Guzzle repo that actually says it uses the `defaults` subarray. The phpdoc merely says that any request option can be passed as a default. `defaults` isn't a request option: `headers` is.

I think the unit tests would have failed if this wasn't the correct way to set the defaults, by the way.

Comment (dcu) [inline lib/Authy/AuthyApi.php]: you're right Vic, it doesn't work. 

I'll merge your changes Thanks!

Comment (vicdelfant) [inline lib/Authy/AuthyApi.php]: You're welcome! :)

**Comment by adrienbrault** [comment]
This needs rebase

**Comment by serargz** [comment]
@vicdelfant Thanks for taking the time to work on this, and sorry for the ridiculously late response. We have merged https://github.com/authy/authy-php/pull/18 which contains your changes. Hopefully we'll be giving more love to this repo in the future.
